crypto/storage/azure/keyvault.go

```diff
@@ -0,0 +1,276 @@
+		client:           a.client,
+		timeOut:          a.timeOut,
+		keyName:          keyName,
+		publicKey:        publicKey,
+		signingAlgorithm: signingAlgorithm,
+	}, nil
+}
+
+func (a keyvault) PrivateKeyExists(ctx context.Context, kid string) (bool, error) {
+	_, err := a.getPrivateKey(ctx, keyIDToKeyName(kid))
+	if errors.Is(err, spi.ErrNotFound) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (a keyvault) DeletePrivateKey(ctx context.Context, kid string) error {
+	_, err := a.client.DeleteKey(ctx, keyIDToKeyName(kid), nil)
+	responseError := new(azcore.ResponseError)
+	if errors.As(err, &responseError) && responseError.StatusCode == http.StatusNotFound {
+		return spi.ErrNotFound
+	} else if err != nil {
+		return fmt.Errorf("unable to delete key from Azure Key Vault (name=%s): %w", keyIDToKeyName(kid), err)
+	}
+	return nil
+}
+
+func (a keyvault) getPrivateKey(ctx context.Context, keyName string) (*azkeys.GetKeyResponse, error) {
+	response, err := a.client.GetKey(ctx, keyName, "", nil)
+	responseError := new(azcore.ResponseError)
+	if errors.As(err, &responseError) && responseError.StatusCode == http.StatusNotFound {
+		return nil, spi.ErrNotFound
+	} else if err != nil {
+		// other error
+		return nil, fmt.Errorf("unable to get key from Azure Key Vault (name=%s): %w", keyName, err)
+	}
+	return &response, nil
+}
+
+func (a keyvault) SavePrivateKey(ctx context.Context, kid string, key crypto.PrivateKey) error {
+	// Only used for migrating to a new storage backend, which is not implemented yet for Azure Key Vault
+	return errors.New("SavePrivateKey() is not supported for Azure Key Vault")
+}
+
+func (a keyvault) ListPrivateKeys(ctx context.Context) []string {
+	// Only used for:
+	// - migrating to a new storage backend, which is not implemented yet for Azure Key Vault
+	// - did:nuts DIDs ownership checking, which won't be supported by this Azure Key Vault integration
+	return nil
+}
+
+// parseKey parses an Azure Key Vault key into a crypto.PublicKey and selects the azkeys.SignatureAlgorithm.
+func parseKey(key *azkeys.JSONWebKey) (crypto.PublicKey, azkeys.SignatureAlgorithm, error) {
+	jwkData, _ := json.Marshal(key)
+	keyAsJWK, err := jwk.ParseKey(jwkData)
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to parse key from Azure Key Vault as JWK: %w", err)
+	}
+	var publicKey crypto.PublicKey
+	if err := keyAsJWK.Raw(&publicKey); err != nil {
+		return nil, "", fmt.Errorf("unable to convert key from Azure Key Vault Key to crypto.PublicKey: %w", err)
+	}
+	if !(*key.Kty == azkeys.KeyTypeEC || *key.Kty == azkeys.KeyTypeECHSM) || *key.Crv != azkeys.CurveNameP256 {
+		return nil, "", errors.New("only ES256 keys are supported")
+	}
+	// Code to support all types, supported by Azure Key Vault (kept here for future support):
+	//var signatureAlgorithm azkeys.SignatureAlgorithm
+	//switch *key.Kty {
+	//case azkeys.KeyTypeEC:
+	//	fallthrough
+	//case azkeys.KeyTypeECHSM:
```

Comment:
```suggestion
	//case azkeys.KeyTypeEC, azkeys.KeyTypeECHSM:
```

I believe comma separating values is preferred over using empty `fallthrough` cases

Reply:
Removed